### DISCUSSION TITLE
Setting Clarification

### DISCUSSION THREAD

**Discussion by Ichisich** [RamonUnch/AltDrag#451] [Q&A]
In case someone has a question regarding a setting which needs clarification.

Personally, I always wondered why `RezTimer` acts so differently depending on my monitor's refresh rate.
The same monitor with 75Hz acts jittery, while with 60Hz the result is often more smooth than even native dragging.
The .ini reads:
"[...]move the window only when the timestamp of a mouse event is different from the previous one, **every 16ms more or less**."
Considering 16,6 ms is the refresh cycle of 60Hz, is the interval hard-coded?

**Comment by RamonUnch** [comment]
Each mouse event AltSnap receives is contains a time stamp, the time stamp is in millisecond but is not millisecond accurate the accuracy is approximately 16ms. This value is not decided by AltSnap but by Windows and it seems to have been this way since the early Windows NT days.
The real timestamp is thus rounded. This mean that events that are close to each-other will have exactly the same timestamp. from my experience it seems that the steps are 15-16ms apart.
For example with my 500Hz mouse:
```
List of mouse event timestamps in milliseconds, (when moving the mouse)
1598980500
1598980500
1598980500
1598980500
1598980500
1598980500
1598980500
1598980500
1598980515
1598980515
1598980515
1598980515
1598980515
1598980515
1598980515
1598980515
1598980531
1598980531
1598980531
1598980531
1598980531
1598980531
1598980531
1598980531
```
We see the same timestamp repeat 8 times, then the timer increments 15 or 16ms and again 

So when you set `RezTimer=1` AltSnap will ignore all events that have the same timestamp than a previous one, reducing the effective mouse rate to 16ms. As you say this is the reason you have problems with this option when using a screen with a refresh-rate higher than 60Hz.

I think I can improve the Move/Resize rates to combine and with RezTimer and get the best of both words.

**Reply by Ichisich** [reply]
Interesting, I read that the NT kernel has a maximum timer resolution of just 1ms and got an upgrade to 0,5ms with Windows 10+.
[https://stackoverflow.com/questions/3141556/how-to-setup-timer-resolution-to-0-5-ms#3172064](https://stackoverflow.com/questions/3141556/how-to-setup-timer-resolution-to-0-5-ms#3172064)
[https://yuzu-emu.org/entry/yuzu-progress-report-mar-2023/](https://yuzu-emu.org/entry/yuzu-progress-report-mar-2023/)


But no idea if this can be applicable to the input system.

**Reply by RamonUnch** [reply]
It might be possible however this kind of high resolution timer is designed for video-games and not desktop applications.
If AltSnap were to change this resolution, it would apply system-wide and might interfere with other programs. In addition it would reduce overall system performances and increase power consumption (probably not much though).
I do use a high-resolution timer for the `RefreshRate` option but only for the shortest possible duration to minimize system impact.

**Reply by Ichisich** [reply]
The most efficient approach would be to read the monitor refresh rate and then adapt the intervals.
In case of 75Hz, this would be roughly 13ms instead of 16ms.
But if Windows hard-codes 16ms, there is probably no simple solution to this.
